pretyping/vnorm.ml

```diff
@@ -33,8 +32,8 @@ let e_whd_all = Reductionops.clos_whd_flags CClosure.all
 
 let crazy_type =  mkSet
 
-let decompose_prod env t =
-  let (name,dom,codom) = destProd (whd_all env t) in
+let decompose_prod env sigma t =
+  let (name,dom,codom) = destProd (EConstr.Unsafe.to_constr (e_whd_all env sigma (EConstr.of_constr t))) in
```

Comment:
The Unsafe.to_constr here is not pretty but trying to do without it I was fighting the types more than seemed worth doing.